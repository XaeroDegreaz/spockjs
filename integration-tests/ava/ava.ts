import { resolve } from 'path';

import { modulePath, runWithTypescriptJit } from '../utils';

const avaCli = resolve(modulePath('ava'), 'cli.js');

export const runAva = (cwd: string) => {
  const { status, stdout } = runWithTypescriptJit(
    [avaCli, '--serial', '--no-cache', '--verbose'],
    { cwd },
  );

  return {
    status,
    stdout: stdout
      .toString()
      .replace(/✔|√|✖|×/g, ' ') // remove OS-dependent symbols
      .replace(/[^ ]+ava\.js:/, '<FILEPATH HIDDEN>:'), // remove absolute path
  };
};
