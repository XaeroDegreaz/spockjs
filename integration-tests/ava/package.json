{
  "ava": {
    "files": ["ava.js"],
    "failWithoutAssertions": false,
    "babel": {
      "testOptions": {
        "plugins": ["../../packages/babel-plugin-spock/src/index.ts"]
      }
    }
  },
  "babel": {
    "presets": [["@babel/preset-env", { "targets": { "node": 6 } }]]
  }
}
