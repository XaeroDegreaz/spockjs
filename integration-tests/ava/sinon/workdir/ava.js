import test from 'ava';
import * as sinon from 'sinon';

test('stubbing', t => {
  const s = sinon.stub();
  stub: s() >> 42;
  t.truthy(s() === 42);
});

test('mocking', t => {
  const obj = { method: () => {} };

  const mock = sinon.mock(obj);
  mock: 2 * mock.method();

  obj.method();
  obj.method();

  verify: mock;
});

test('failing verification', t => {
  const obj = { method: () => {} };

  const mock = sinon.mock(obj);
  mock: 2 * mock.method();

  obj.method();

  verify: mock;
});
