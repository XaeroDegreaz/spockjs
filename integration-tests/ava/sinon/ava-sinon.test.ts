import { resolve } from 'path';

import { runAva } from '../ava';

// mark implicit dependencies for jest
() => require('./workdir/ava.js') && require('./workdir/package.json');
() => require('@spockjs/preset-runner-ava');
() => require('@spockjs/preset-sinon-mocks');

const cwd = resolve(__dirname, 'workdir');

test('interaction blocks work', () => {
  const { status, stdout } = runAva(cwd);

  expect: status === 1;
  expect(
    stdout.replace(/^.+\.(j|t)s(:[0-9]+){1,2}.*(\r\n|\r|\n)/gm, ''),
  ).toMatchSnapshot();
});
