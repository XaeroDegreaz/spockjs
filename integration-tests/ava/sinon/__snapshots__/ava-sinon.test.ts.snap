// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interaction blocks work 1`] = `
"
    stubbing
    mocking
    failing verification Error thrown in test

  1 test failed

  failing verification


  Error thrown in test:

  ExpectationError (Error) {
    message: 'Expected method([...]) twice (called once)',
  }

  t <FILEPATH HIDDEN>:22:35)

"
`;
